tests/templates/test_subroutines/test_hilbert_schmidt.py

```diff
@@ -0,0 +1,402 @@
+
+    def test_hs_decomposition_1_qubit(self):
+        """Test if the HS operation is correctly decomposed for 1 qubit."""
+        with qml.tape.QuantumTape(do_queue=False) as U:
+            qml.Hadamard(wires=0)
+
+        def v_circuit(params):
+            qml.RZ(params[0], wires=1)
+
+        op = qml.HilbertSchmidt([0.1], v_function=v_circuit, v_wires=[1], u_tape=U)
+
+        with qml.tape.QuantumTape() as tape_dec:
+            op.decomposition()
+
+        expected_operations = [
+            qml.Hadamard(wires=[0]),
+            qml.CNOT(wires=[0, 1]),
+            qml.Hadamard(wires=[0]),
+            qml.RZ(-0.1, wires=[1]),
+            qml.CNOT(wires=[0, 1]),
+            qml.Hadamard(wires=[0]),
+        ]
+        for i, j in zip(tape_dec.operations, expected_operations):
+            assert i.name == j.name
+            assert i.data == j.data
+            assert i.wires == j.wires
+
+    def test_hs_adjoint_method(self):
+        """Test the adjoint method of the HS operation for 1 qubit."""
+
+        with qml.tape.QuantumTape(do_queue=False) as U:
+            qml.RX(0.3, wires=0)
+
+        def v_circuit(params):
+            qml.RZ(params[0], wires=1)
+
+        adj = qml.HilbertSchmidt([0.1], v_function=v_circuit, v_wires=[1], u_tape=U).adjoint()
+        operations = adj.expand().operations
+
+        expected_operations = [
+            qml.Hadamard(wires=[0]),
+            qml.CNOT(wires=[0, 1]),
+            qml.RZ(0.1, wires=[1]),
+            qml.RX(-0.3, wires=[0]),
+            qml.CNOT(wires=[0, 1]),
+            qml.Hadamard(wires=[0]),
+        ]
+
+        for i, j in zip(operations, expected_operations):
+            assert i.name == j.name
+            assert i.data == j.data
+            assert i.wires == j.wires
+
+    def test_hs_decomposition_2_qubits(self):
+        """Test if the HS operation is correctly decomposed for 2 qubits."""
+        with qml.tape.QuantumTape(do_queue=False) as U:
+            qml.SWAP(wires=[0, 1])
+
+        def v_circuit(params):
+            qml.RZ(params[0], wires=2)
+            qml.CNOT(wires=[2, 3])
+
+        op = qml.HilbertSchmidt([0.1], v_function=v_circuit, v_wires=[2, 3], u_tape=U)
+
+        with qml.tape.QuantumTape() as tape_dec:
+            op.decomposition()
+
+        expected_operations = [
+            qml.Hadamard(wires=[0]),
+            qml.Hadamard(wires=[1]),
+            qml.CNOT(wires=[0, 2]),
+            qml.CNOT(wires=[1, 3]),
+            qml.SWAP(wires=[0, 1]),
+            qml.RZ(-0.1, wires=[2]),
+            qml.CNOT(wires=[2, 3]),
+            qml.CNOT(wires=[1, 3]),
+            qml.CNOT(wires=[0, 2]),
+            qml.Hadamard(wires=[0]),
+            qml.Hadamard(wires=[1]),
+        ]
+
+        for i, j in zip(tape_dec.operations, expected_operations):
+            assert i.name == j.name
+            assert i.data == j.data
+            assert i.wires == j.wires
+
+    def test_hs_adjoint_method_2_qubits(self):
```

Comment:
Maybe good to add a variation with custom wire labels too (e.g., strings).

Reply:
Good idea, I will add it!